src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -63,6 +64,22 @@ public class TestSetupHooks {
   public static String spannerTargetTable = StringUtils.EMPTY;
   public static boolean firstSpannerTestFlag = true;
 
+  @Before(order = 1)
+  public static void overrideServiceAccountFilePathIfProvided() {
+    if (beforeAllFlag) {
+      String serviceAccountType = System.getenv("SERVICE_ACCOUNT_TYPE");
+      if (serviceAccountType != null) {
+        if (serviceAccountType.equalsIgnoreCase("FilePath")) {
+          String serviceAccountFilePath = System.getenv("SERVICE_ACCOUNT_FILE_PATH");
+          if (!(serviceAccountFilePath == null) && !serviceAccountFilePath.equalsIgnoreCase("auto-detect")) {
+            PluginPropertyUtils.addPluginProp("serviceAccountFilePath", serviceAccountFilePath);
```

Comment:
Why is this step needed if we are reading directly from env variables?
https://github.com/cdapio/cdap-e2e-tests/blob/develop/src/main/java/io/cdap/e2e/pages/actions/CdfPluginPropertiesActions.java#L591

Reply:
This is required for macro runtime arguments. 
https://github.com/data-integrations/google-cloud/blob/2774b692b9d105595b87b2942b8e59e746d0f780/src/e2e-test/features/bigquery/sink/GCSToBigQuery_WithMacro.feature#L35

Reply:
I think the macro scenario needs to be modified, what if service account type is set as "JSON" instead of "filePath". It will fail, right?

Reply:
We have separate macro tests with Service Account Type as macro param

https://github.com/data-integrations/google-cloud/pull/937

Reply:
What about the existing macro tests? They will fail if serviceAccountType is set as `JSON` in environment variable, right?

Reply:
Updated macro tests.